editor/src/components/canvas/controls/select-mode/absolute-resize-control.tsx

```diff
@@ -0,0 +1,334 @@
+import { useColorTheme } from '../../../../uuiui'
+import { EditorDispatch } from '../../../editor/action-types'
+import { setResizeOptionsTargetOptions } from '../../../editor/actions/action-creators'
+import { useEditorState, useRefEditorState } from '../../../editor/store/store-hook'
+import CanvasActions from '../../canvas-actions'
+import {
+  CSSCursor,
+  DirectionAll,
+  DirectionHorizontal,
+  DirectionVertical,
+  EdgePosition,
+  resizeDragState,
+  updateResizeDragState,
+} from '../../canvas-types'
+import { getOriginalFrames } from '../../canvas-utils'
+import { windowToCanvasCoordinatesGlobal } from '../../dom-lookup'
+import { useBoundingBox } from '../bounding-box-hooks'
+
+interface AbsoluteResizeControlProps {
+  localSelectedElements: Array<ElementPath>
+}
+
+export const AbsoluteResizeControl = React.memo<AbsoluteResizeControlProps>((props) => {
+  const localSelectedElements = props.localSelectedElements
+  const allSelectedElementsAbsolute = useEditorState((store) => {
+    return (
+      localSelectedElements.length > 0 &&
+      localSelectedElements.every((path) => {
+        return (
+          MetadataUtils.findElementByElementPath(store.editor.jsxMetadata, path)
+            ?.specialSizeMeasurements.position === 'absolute'
+        )
+      })
+    )
+  }, 'AbsoluteResizeControl allSelectedElementsAbsolute')
+
+  const absoluteElements = allSelectedElementsAbsolute ? localSelectedElements : []
+
+  const controlRef = useBoundingBox(absoluteElements, (ref, boundingBox) => {
+    ref.current.style.left = boundingBox.x + 'px'
+    ref.current.style.top = boundingBox.y + 'px'
+    ref.current.style.width = boundingBox.width + 'px'
+    ref.current.style.height = boundingBox.height + 'px'
+  })
+
+  const leftRef = useBoundingBox(absoluteElements, (ref, boundingBox) => {
+    ref.current.style.height = boundingBox.height + 'px'
+  })
+  const topRef = useBoundingBox(absoluteElements, (ref, boundingBox) => {
+    ref.current.style.width = boundingBox.width + 'px'
+  })
+  const rightRef = useBoundingBox(absoluteElements, (ref, boundingBox) => {
+    ref.current.style.left = boundingBox.width + 'px'
+    ref.current.style.height = boundingBox.height + 'px'
+  })
+
+  const bottomRef = useBoundingBox(absoluteElements, (ref, boundingBox) => {
+    ref.current.style.top = boundingBox.height + 'px'
+    ref.current.style.width = boundingBox.width + 'px'
+  })
+
+  const topLeftRef = useBoundingBox(absoluteElements, NO_OP)
+  const topRightRef = useBoundingBox(absoluteElements, (ref, boundingBox) => {
+    ref.current.style.left = boundingBox.width + 'px'
+  })
+  const bottomLeftRef = useBoundingBox(absoluteElements, (ref, boundingBox) => {
+    ref.current.style.top = boundingBox.height + 'px'
+  })
+  const bottomRightRef = useBoundingBox(absoluteElements, (ref, boundingBox) => {
+    ref.current.style.left = boundingBox.width + 'px'
+    ref.current.style.top = boundingBox.height + 'px'
+  })
```

Comment:
All of these refs look like they're setting the wrong values on the style object, e.g. the `leftRef` (which I'm assuming is for the left edge control) is setting the height?

Reply:
Oh, sorry, ignore this! :facepalm: